resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -164,6 +165,9 @@ private[spark] class Client(
         }
       } finally {
         kubernetesResourceCleaner.deleteAllRegisteredResourcesFromKubernetes(kubernetesClient)
+        if (loggingWatch.nonEmpty) {
+          loggingWatch.get.shutdown()
```

Comment:
I would expect `close` to handle this and that we would catch this since we wrap the call in a `tryWithResource1 - is this not the case?